internal/controller/runtime/fsm/runtime_state.go

```diff
@@ -0,0 +1,21 @@
+package fsm
+
+import (
+	gardener_api "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+)
+
+// the state of controlled system (k8s cluster)
+type systemState struct {
+	instance imv1.Runtime
+	snapshot imv1.RuntimeStatus
+	shoot    *gardener_api.Shoot
+}
+
+func (s *systemState) saveRuntimeStatus() {
```

Comment:
What is the purpose of this function? I can't see where the `snapshot` is used.

Reply:
Cache temporarily Status when reconcilation starts to compare it for eventually emitting Event. It is required.